elephant/statistics.py

```diff
@@ -799,23 +809,416 @@ def complexity_pdf(spiketrains, binsize):
+        """
+        Get an epoch object of the complexity distribution with `spread` = 0
+        """
+        left_edges = self.time_histogram.times
+        durations = self.bin_size * np.ones(self.time_histogram.shape)
+        if self.sampling_rate:
+            # ensure that spikes are not on the bin edges
+            bin_shift = .5 / self.sampling_rate
+            left_edges -= bin_shift
+        else:
+            warnings.warn('No sampling rate specified. '
+                          'Note that using the complexity epoch to get '
+                          'precise spike times can lead to rounding errors.')
+
+        # ensure that an epoch does not start before the minimum t_start
+        min_t_start = min([st.t_start for st in self.input_spiketrains])
+        if left_edges[0] < min_t_start:
+            left_edges[0] = min_t_start
+            durations[0] -= bin_shift
+
+        epoch = neo.Epoch(left_edges,
+                          durations=durations,
+                          array_annotations={
+                              'complexity':
+                              self.time_histogram.magnitude.flatten()})
+        return epoch
+
+    def _epoch_with_spread(self):
+        """
+        Get an epoch object of the complexity distribution with `spread` > 0
+        """
+        bst = conv.BinnedSpikeTrain(self.input_spiketrains,
+                                    binsize=self.bin_size,
+                                    tolerance=self.tolerance)
+
+        if self.binary:
+            binarized = bst.to_sparse_bool_array()
+            bincount = np.array(binarized.sum(axis=0)).squeeze()
+        else:
+            bincount = np.array(bst.to_sparse_array().sum(axis=0)).squeeze()
+
+        i = 0
+        complexities = []
+        left_edges = []
+        right_edges = []
+        while i < len(bincount):
+            current_bincount = bincount[i]
+            if current_bincount == 0:
+                i += 1
+            else:
+                last_window_sum = current_bincount
+                last_nonzero_index = 0
+                current_window = bincount[i:i + self.spread + 1]
+                window_sum = current_window.sum()
+                while window_sum > last_window_sum:
+                    last_nonzero_index = np.nonzero(current_window)[0][-1]
+                    current_window = bincount[i:
+                                              i + last_nonzero_index
+                                              + self.spread + 1]
+                    last_window_sum = window_sum
+                    window_sum = current_window.sum()
+                complexities.append(window_sum)
+                left_edges.append(
+                    bst.bin_edges[i].magnitude.item())
+                right_edges.append(
+                    bst.bin_edges[
+                        i + last_nonzero_index + 1
+                    ].magnitude.item())
+                i += last_nonzero_index + 1
```

Comment:
this block will be the bottleneck in understanding the functionality in the future.
I do see that you have spent time thinking on appropriate names that could "tell a story" and yet...
I don't have an alternative suggestion. Keeping it in the current state for now.